File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/testClasses/MotorTest.java
package org.firstinspires.ftc.teamcode.testClasses;

import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
import com.qualcomm.robotcore.hardware.CRServo;
import com.qualcomm.robotcore.hardware.CRServoImplEx;
import com.qualcomm.robotcore.hardware.DcMotor;
import com.qualcomm.robotcore.hardware.Servo;

import org.firstinspires.ftc.teamcode.autoRes.commands.DriveCommand;
import org.firstinspires.ftc.teamcode.subsystems.Chassis;
import org.firstinspires.ftc.teamcode.subsystems.Robot;
/*
@TeleOp(name = "Motor Test", group = "Teleop")
public class MotorTest extends LinearOpMode {
    long timeTo = System.currentTimeMillis();

    @Override
    public void runOpMode() {
        Robot robot = new Robot(hardwareMap);
        Chassis chassis = new Chassis(hardwareMap);
        //chassis.reverseMotors(new DcMotor[]{chassis.frontRight, chassis.backRight});
        robot.setChassis(chassis);
        DcMotor motor = hardwareMap.dcMotor.get("test");
        motor.setZeroPowerBehavior(DcMotor.ZeroPowerBehavior.BRAKE);
        waitForStart();
        while (opModeIsActive()) {
            telemetry.addData("hit", robot.chassis.backRight.getCurrentPosition());
            if (gamepad1.a) {
                chassis.runRotations(.5, .25);
            }
            if (gamepad1.b) {
                robot.arm.run(3);
            }
            if (gamepad1.x) {
                robot.intake.setMainToHighPosition();
            }
            if (gamepad1.y) {
                robot.arm.setArm(4);
                chassis.runRotations(.5, -.25);
            }
            //Other unit test code if you want
            telemetry.update();
            }
            }
 */

File: TeamCode/src/onlyTeamCode/java/org/firstinspires/ftc/teamcode/test/MotorTest.kt
package org.firstinspires.ftc.teamcode.test

import com.google.firebase.database.DatabaseReference
import com.qualcomm.robotcore.eventloop.opmode.TeleOp
import org.firstinspires.ftc.teamcode.MOEStuff.MOEOpmodes.MOETeleOp
import org.firstinspires.ftc.teamcode.utilities.addData

@TeleOp(name = "MotorTest")
class MotorTest : MOETeleOp() {
    override fun run() {
        while (opModeIsActive()) {
            loopStuff()
        }
    }

    override fun getCustomRef(ref: DatabaseReference): DatabaseReference? {
        return ref
    }

    override fun initOpMode() {
        telemetry.addData("testagain")
        robot.odometry.servos.initServosUp()
    }

    private fun loopStuff() {
        robot.chassis.setPower(FLP = 1.0, FRP = 0.0, BLP = 0.0, BRP = 0.0)
    }
}